File: data_structures_java/MyQueue.java
public class MyQueue<T> {
	
	private int front;
	private int rear;
	private int max;
	private T[] arr;

	MyQueue() {
		this(5);
	}
	MyQueue(int max) {
		this.max = max;
		front = -1;
		rear = -1;
		arr = (T[]) new Object[max];
	}
	public boolean isEmpty() {
		return (front == -1 && rear == -1);
	}
	public boolean isFull() {
		return (rear - front == max - 1);
	}
	void enQueue(T item) {
		if (isEmpty()) {
			front = 0;
			rear = 0;
			arr[front] = item;
		} else if (!isFull()){
			rear += 1;
			arr[rear] = item;
		}
	}
	void deQueue() {
		if (front == rear) {
			front = -1;
			rear = -1;
		} else {
			front += 1;
		}
	}
	void printQueue() {
		if (!isEmpty()){
			for (int i=front; i<=rear; i++) {
				System.out.println(arr[i]);
			}
		}
	}

	public static void main(String[] args) {
		MyQueue<String> mq = new MyQueue<String>();

		mq.enQueue("34");
		mq.enQueue("14");
		mq.enQueue("394");
		mq.enQueue("4");
		mq.enQueue("77");
		mq.enQueue("12");   // won't be added since out of space
	
		mq.deQueue();
		mq.deQueue();
		mq.deQueue();
	
		mq.printQueue();

	}
}

File: data_structures_js/myqueue.js
class MyQueue {
	
	constructor(max = 5) {
		this.max = max;
		this.front = -1;
		this.rear = -1;
		this.arr = [];
	}
	isEmpty() {
		return (this.front == -1 && this.rear == -1);
	}
	isFull() {
		return (this.rear - this.front == this.max - 1);
	}
	enQueue(item) {
		if (this.isEmpty()) {
			this.front = 0;
			this.rear = 0;
			this.arr[this.front] = item;
		} else if (!this.isFull()){
			this.rear += 1;
			this.arr[this.rear] = item;
		}
	}
	deQueue() {
		if (this.front == this.rear) {
			this.front = -1;
			this.rear = -1;
		} else {
			this.front += 1;
		}
	}
	printQueue() {
		if (!this.isEmpty()){
			for (let i=this.front; i<=this.rear; i++) {
				console.log(this.arr[i])
			}
		}
	}
}

if (require.main == module) {

	mq = new MyQueue();

	mq.enQueue("34");
	mq.enQueue("14");
	mq.enQueue("394");
	mq.enQueue("4");
	mq.enQueue("77");
	mq.enQueue("12");   // won't be added since out of space

	mq.deQueue();
	mq.deQueue();
	mq.deQueue();

	mq.printQueue();
}